src/lib/Polykey.ts

```diff
@@ -17,21 +18,22 @@ class Polykey {
     polykeyPath: string = `${os.homedir()}/.polykey`,
     fileSystem: typeof fs,
     keyManager?: KeyManager,
-    vaultManager?: VaultManager,
     peerManager?: PeerManager,
+    vaultManager?: VaultManager,
   ) {
     this.polykeyPath = polykeyPath;
 
     // Set key manager
     this.keyManager = keyManager ?? new KeyManager(this.polykeyPath, fileSystem);
 
-    // Set or Initialize vaultManager
-    this.vaultManager = vaultManager ?? new VaultManager(this.polykeyPath, fileSystem, this.keyManager);
-
     // Initialize peer store and peer discovery classes
-    this.peerManager = peerManager ?? new PeerManager(this.polykeyPath, fileSystem, this.keyManager, this.vaultManager);
+    this.peerManager = peerManager ?? new PeerManager(this.polykeyPath, fileSystem, this.keyManager);
+
+    // Set or Initialize vaultManager
+    this.vaultManager =
+      vaultManager ?? new VaultManager(this.polykeyPath, fileSystem, this.keyManager, this.peerManager);
   }
 }
 
 export default Polykey;
-export { KeyManager, VaultManager, PeerManager, PolykeyAgent, PolykeyClient };
+export { KeyManager, VaultManager, PeerManager, PolykeyAgent, PolykeyClient, PeerInfo };
```

Comment:
Are we expecting users of the library to construct `new PeerInfo`?

Reply:
no, the agent code uses PeerInfo in its codebase and there is already a command in webpack to stop compiling the lib code whenever there is an import from `../lib/Polykey`. You can think of it as a single point of interface that the agent can use instead of importing from lets say `../lib/peers/PeerInfo`

Reply:
or maybe we just don't care if webpack compiles `PeerInfo` twice into both agent. I don't think there is any other way to do it and stop this behaviour. Probably won't be a concern when we move away from webpack.